### PR TITLE
metafeed/announce is signed by meta feed

### PR DESCRIPTION
**Pull request by staltz**
For issue https://github.com/ssb-ngi-pointer/ssb-meta-feeds-spec/issues/27

### CODE REVIEW

**Comment by staltz** [comment]
Oh, I should probably update validation to check these sigs.

**Comment by staltz** [comment]
> Oh, I should probably update validation to check these sigs.

Oh well actually `metafeed/announce` is published on the main feed, and we only need to validate remote peer's `metafeed/announce` msgs, which should happen in ssb-replication-scheduler.

So this PR is ready to go I guess

**Comment by staltz** [comment]
Or hey I can just make a new function, `validateAnnounce(msg)`. Yes

File: validate.js [hunk 1]
```diff
@@ -173,5 +182,39 @@ function validateHmacKey(hmacKey) {
...
+        `because content.metafeed is not a bendy butt feed: ${metaFeedId}`
+    )
+  }
+
+  const { data } = SSBURI.decompose(metaFeedId)
+  const ed25519Public = `${data}.ed25519`
+  if (!ssbKeys.verifyObj(ed25519Public, msg.value.content)) {
+    return new Error(
+      `metafeed/announce ${msg.key} is invalid ` +
+        `because content is not a signed by the meta feed: ${msg.value.content}`
```
Comment (arj03) [inline validate.js:213]: => because content is not signed by the meta feed: ${msg.value.content}

File: test/query.js [hunk 2]
```diff
@@ -123,7 +123,7 @@ test('seed', (t) => {
 })
 
 test('announce', (t) => {
-  messages.getContentAnnounce(metafeedKeys.id, (err, content) => {
+  messages.getContentAnnounce(metafeedKeys, (err, content) => {
```
Comment (arj03) [inline test/query.js:126]: There must have been a bug here before?

**Review by arj03 — APPROVED** [review]
Just found that minor spelling thing, otherwise looks good

File: test/query.js (hunk 2 above)
Comment (staltz) [reply test/query.js:126]: Yes there was. The checking is done correctly in `test/messages.js` but in this file we want to test the query APIs so we just used `getContentAnnounce` to create the inputs, but we didn't validate the input. Anyway whatever :)

**Pull request merged**